### PR TITLE
Token refresh

### PR DESCRIPTION
**Pull request by kevincasey1222**
Hi @mknoedel !

Here is the code to allow the GraphQL client to update its token when it is about to expire.

I built this off the the Updated mfaEnabled prop to undefined when null PR, so this includes those changes too, and a few misc fixes I discovered along the way.

### CODE REVIEW

File: src/steps/reposecrets.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export const repoSecretSteps: IntegrationStep<IntegrationConfig>[] = [
         targetType: GITHUB_ORG_SECRET_ENTITY_TYPE,
       },
     ],
-    dependsOn: ['fetch-org-secrets'],
```
Comment (mknoedel) [inline src/steps/reposecrets.ts:132]: Do we still need this step to depend on `fetchhrepos`?

File: src/steps/envsecrets.ts [hunk 2]
```diff
@@ -144,7 +144,11 @@ export const envSecretSteps: IntegrationStep<IntegrationConfig>[] = [
         targetType: GITHUB_REPO_SECRET_ENTITY_TYPE,
       },
     ],
-    dependsOn: ['fetch-environments', 'fetch-repo-secrets'],
+    dependsOn: [
+      'fetch-environments',
+      'fetch-repo-secrets',
+      'fetch-org-secrets',
```
Comment (mknoedel) [inline src/steps/envsecrets.ts:150]: good catch with this one, this might have been causing some problems as well.

**Review by mknoedel — APPROVED** [review]
Looks great thanks!

File: src/steps/reposecrets.ts (hunk 1 above)
Comment (kevincasey1222) [reply src/steps/reposecrets.ts:132]: Yeah, still needs that, so populate repo tags.

**Pull request merged**